source/base/subscriptor.cc

```diff
@@ -175,7 +158,6 @@ Subscriptor::subscribe(const char *id) const
 #ifdef DEBUG
   if (object_info == 0)
     object_info = &typeid(*this);
-  Threads::Mutex::ScopedLock lock (subscription_lock);
   ++counter;
```

Comment:
That was how I read your original comment (subtext in Internet comments is hard); my point is that I don't really understand the documentation here either.